### DISCUSSION TITLE
Updates to DCDB interfaces for VBI

### DISCUSSION THREAD

**Discussion by brian-r-calder** [CCOMJHC/OpenVBI#1] [Ideas]
At the IHO CSBWG meeting last week in Monaco, the DCDB people announced that they have developer time this summer to built new features, or adjust current features, in the implementation of the VBI stores.  They already have plans for some modifications (e.g., better access to metadata by unique ID within the point store version of the DCDB holdings), but are open to community suggestions.

There will probably be a number of submissions, but if we can put together reasoned suggestions for what to modify (and a priority), I think it might be possible to get some forward momentum on getting things fixed.

The floor is therefore open ...

**Comment by ShaulSolomon** [comment]
I have a few ideas, primarily focused on advanced queries and APIs that I think is interesting, but I am not sure what is currently available and what isn't. Could someone direct me to more information?

**Reply by brian-r-calder** [reply]
Try https://www.ncei.noaa.gov/iho-data-centre-digital-bathymetry and then click on"crowdsourced bathymetry" and "Access CSB Data" to find the links to more information.  Unfortunately, it's all one page, so I can't give a more direct URL.

**Comment by ShaulSolomon** [comment]
Requests (order is based off priority):

###  A
```
"collection_date": {
            "type": "string",
            "description": "date on which the data were collected. Format: YYYY-MM-DD"
          },
"archive_date": {
            "type": "string",
            "description": "date on which the data were received at the DCDB. Format: YYYY-MM-DD"
          },
```
Can we allow it to filter out between dates for both collection/archive - with strong preference to `collection_date` ?
Example: `{"collection_date" : ["1970-01-01", "1970-01-02"]}`

### B
```
"provider": {
            "type": "string",
            "description": "trusted node name. only used for csb"
          },
"platform" : {
            "type": "string",
            "description": "platform, aka ship or vessel name"
          }
```
Can we change the types to `list[string]`, so that we can select from a list of possible vessels at once, without having to make several request?

### C
```
"bbox": {
  "type": "string",
  "description": "comma-separated list of geographic coordinates in units of decimal degrees. format: minx, miny, maxx, maxy"
},
```
Can we allow for the bbox be a Shapely Polygon (epsg:4326)?
Example: `POLYGON ((-74.05050806403247 40.5664220341608…`

**Comment by ShaulSolomon** [comment]
Another Request.

Helper queries to extract crucial metadata from the databases.

Examples: 
1) For a given spatial and temporal range, please list total number of data points available, and the name of all "providers" and "platforms".

2) For each provider, get a list of all platforms.

**Comment by colinthomson-consultingservices** [comment]
A request I talked with Georgie awhile back.

Allow a customized URL to be constructed and shared that would allow you to use the data viewer to point directly to providers or platforms using the standard filters already in place. This would allow the viewer to be used to share data uploaded to a specific audience (ie to the contributing vessels or organizations)

**Comment by pdebroisse** [comment]
Instead of going to the API to get file names and then going to the S3 bucket to retrieve data, a GeoPackage (or similar) file at the top of the S3 bucket allowing it to be more easily searched would be helpful. 

There are also metadata for vessels available in the GeoJSON files that is not available in the CSV files in the S3 bucket. Some sort of dictionary file at the top of the bucket with this metadata associated to vessel UUIDs would be helpful for some calculations.
